ethcore/src/engines/tendermint/mod.rs

```diff
@@ -101,6 +101,61 @@ pub struct Tendermint {
 	validators: Box<ValidatorSet>,
 }
 
+struct EpochVerifier {
+	subchain_validators: SimpleList,
+	recover: Box<Fn(&Signature, &Message) -> Result<Address, Error> + Send + Sync>,
```

Comment:
@feynmanliang 

Try
```rust
struct EpochVerifier<F> 
    where F: Fn(&Signature, &Message) -> Result<Address, Error> + Send + Sync 
{
    subchain_validators: SimpleList,
    recover: F,
}
```

and just don't box the closures when you instantiate `EpochVerifier`.